warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RangeCoalescingVisitor.java

```diff
@@ -137,8 +99,9 @@ protected JexlNode coalesceBoundedRanges(Map<LiteralRange<?>,List<JexlNode>> ran
         
         // If we had no other nodes than this bounded range, we can strip out the original parent
         if (newNode.jjtGetNumChildren() == 1) {
-            newNode.jjtGetChild(0).jjtSetParent(newNode.jjtGetParent());
-            return newNode.jjtGetChild(0);
+            JexlNode child = newNode.jjtGetChild(0);
+            JexlNodes.replaceChild(newNode.jjtGetParent(), newNode, child);
```

Comment:
This reparenting is also not needed as reparenting is handled by the base RebuildingVisitor.  

I tested this change locally as well, and tests still pass perfectly if you simply return newNode.jjtGetChild(0).

Reply:
Fixed.